msal/oauth2cli/authcode.py

```diff
@@ -69,14 +55,16 @@ def browse(auth_uri):
     logger.info("Please open a browser on THIS device to visit: %s" % auth_uri)
     controller.open(auth_uri)
 
+
 class AuthCodeReceiver(BaseHTTPRequestHandler):
     def do_GET(self):
         # For flexibility, we choose to not check self.path matching redirect_uri
         #assert self.path.startswith('/THE_PATH_REGISTERED_BY_THE_APP')
         qs = parse_qs(urlparse(self.path).query)
         if qs.get('code'):  # Then store it into the server instance
-            ac = self.server.authcode = qs['code'][0]
-            self._send_full_response('Authcode:\n{}'.format(ac))
+            self.server.auth_code = qs['code'][0]
+            self.server.state = qs.get('state', [None])[0]
```

Comment:
Is it possible to have the initial state pre-polulated inside `self.server`, so that we can check the state here and only accept a `code` when the `state` matches?